pkg/hubble/relay/observer/observer.go

```diff
@@ -218,3 +221,86 @@ func aggregateErrors(
 	}()
 	return aggregated
 }
+
+func sendFlowsResponse(ctx context.Context, stream observerpb.Observer_GetFlowsServer, sortedFlows <-chan *observerpb.GetFlowsResponse) error {
+	for {
+		select {
+		case flow, ok := <-sortedFlows:
+			if !ok {
+				return nil
+			}
+			if err := stream.Send(flow); err != nil {
+				return err
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+func newFlowCollector(g *errgroup.Group, req *observerpb.GetFlowsRequest, opts options) *flowCollector {
+	fc := &flowCollector{
+		g:   g,
+		log: opts.log,
+		ocb: opts.ocb,
+
+		req: req,
+
+		connectedNodes: map[string]struct{}{},
+	}
+	return fc
+}
+
+type flowCollector struct {
+	g   *errgroup.Group
```

Comment:
AFAIK it is idomatic to avoid context to escape the scope of the func where it is created, and I'd think the same reasoning apply for `errgroup`. Can we give the errgroup to `collect()` instead as it is the only place where it's needed?

Reply:
I think you're right. Let's explicitly pass the errgroup to `collect` instead